src/langsmith/agent-server-scale.mdx

```diff
@@ -0,0 +1,361 @@
+
+For example self-hosted configurations, refer to [Example Agent Server configurations for scale](#example-agent-server-configurations-for-scale).
+
+## Scaling for write load
+
+Write load is primarily driven by the following factors:
+
+- Creation of new runs
+- Creation of new checkpoints during run execution
+- Writing to long term memory
+- Creation of new threads
+- Creation of new assistants
+- Deletion of runs, checkpoints, threads, assistants and cron jobs
+
+The following components are primarily responsible for handling write load:
+
+- API server: Handles initial request and persistence of data to the database.
+- Queue worker: Handles the execution of runs.
+- Redis: Handles the storage of ephemeral data about on-going runs.
+- Postgres: Handles the storage of all data, including run, thread, assistant, cron job, checkpointing and long term memory.
+
+### Best practices for scaling the write path
+
+#### Change N_JOBS_PER_WORKER based on assistant characteristics
+
+The default value of [N_JOBS_PER_WORKER](/langsmith/env-var#n-jobs-per-worker) is 10. You can change this value to scale the maximum number of runs that can be executed at a time by a single queue worker based on the characteristics of your assistant.
+
+Some general guidelines for changing N_JOBS_PER_WORKER:
+
+- If your assistant is CPU bounded, the default value of 10 is likely sufficient. You might lower N_JOBS_PER_WORKER if you notice excessive CPU usage on queue workers or delays in run execution.
+- If your assistant is IO bounded, increase N_JOBS_PER_WORKER to handle more concurrent runs per worker.
+
+There is no upper limit to N_JOBS_PER_WORKER. However, Queue workers are greedy when fetching new runs, meaning they will try to pick up as many runs they have available jobs and begin executing them right away. If N_JOBS_PER_WORKER is too high and traffic is burstly, you may see uneven worker utilization and potentially longer run execution times.
+
+#### Avoid synchronous blocking operations
+
+Avoid synchronous blocking operations in your code. If required, set [BG_JOB_ISOLATED_LOOPS](/langsmith/env-var#bg-job-isolated-loops) to True to execute each run in a separate event loop and avoid blocking the main event loop.
+
+#### Minimize redundant checkpointing
+
+Minimize redundant checkpointing by setting [`durability`](/oss/langgraph/durable-execution#durability-modes) to the minimum value necessary to ensure your data is durable.
+
+#### Self-hosted
+
+<Note>
+These settings are only required for self-hosted deployments. By default, Cloud deployments already have these best practices enabled.
+</Note>
+
+###### Enable the use of queue workers
+
+By default, the api server manages the queue and does not use queue workers. You can enable the use of queue workers by setting the `queue.enabled` configuration to `true`.
```

Comment:
```suggestion
By default, the API server manages the queue and does not use queue workers. You can enable the use of queue workers by setting the `queue.enabled` configuration to `true`.
```
When we say API server, are we talking about the Agent Server? (This is more a question for me, btw!)